pkg/compact/compact.go

```diff
@@ -589,3 +597,35 @@ func DownloadBlockDir(ctx context.Context, bucket objstore.Bucket, blockDir, dst
 
 	return nil
 }
+
+// DownloadMeta downloads only meta file from bucket by block ID.
+func DownloadMeta(ctx context.Context, bkt objstore.Bucket, id ulid.ULID) (block.Meta, error) {
+	rc, err := bkt.Get(ctx, path.Join(id.String(), "meta.json"))
+	if err != nil {
+		return block.Meta{}, errors.Wrapf(err, "meta.json bkt get for %s", id.String())
+	}
+	defer rc.Close()
+
+	// Do a full decode/encode cycle to ensure we only print valid JSON.
+	var m block.Meta
+
+	if err := json.NewDecoder(rc).Decode(&m); err != nil {
+		return block.Meta{}, errors.Wrapf(err, "decode meta.json for block %s", id.String())
+	}
+	return m, nil
+}
+
+// ForeachBlockID runs doFn for each block ID in the given bucket.
+func ForeachBlockID(ctx context.Context, bucket objstore.Bucket, doFn func(ulid.ULID) error) error {
+	return bucket.Iter(ctx, "", func(name string) error {
+		if !strings.HasSuffix(name, "/") {
+			return nil
+		}
+		id, err := ulid.Parse(name[:len(name)-1])
+		if err != nil {
+			return nil
+		}
+
+		return doFn(id)
+	})
+}
```

Comment:
`DownloadMeta` and `ForeachBlockID` are not tied to the compactor and seem a bit odd as part of the pkg API. I'd say either make them unexported and duplicate them across packages or move them to a package that's functionally appropriate. `pkg/block` could be appropriate if we are okay with making it depend on `pkg/objstore`.

Reply:
I know, but I wanted objstore and block to be not dependant on each other ^^. Maybe having objstore depent on block is not that bad. Or maybe we need new package